raw/db.py

```diff
@@ -47,25 +48,22 @@ def result(sql, **kwargs):
             result = cur.fetchall()
             cols = cur.keys()
             rows = [dict(zip(cols, row)) for row in result]
+        return rows
 
     except Exception as e:
-        # In case of any exception, capture it and format as result set
-        rows = [{"error": repr(e)}]
+        logging.exception(e)
 
-    return rows
 
```

Comment:
@hkiang01 I really like the addition of logging, but I'm not sure how I feel about not returning anything to the calling function. For starters, it breaks the current API. Not necessarily a huge issue in an 0.x release, but I'm just trying to keep in mind this is intended as a library module, so I try to think of it from the perspective of the calling function. Also, I'm the only known consumer of the library at the moment, and I know I have code that checks for error messages in the result, and this change will break that. Again, maybe that's acceptable, but I'm just trying to start a discussion about whether this is a better interface design as submitted.

Certainly I could consider replacing anything upstream that's like `if "error" in db.result()[0].keys:` with `if db.result() is None:` — the drawback I see is that while an operator can go and review the log, the calling function gets back no information about what went wrong, and hence cannot do anything to inform the user of the larger program. I also don't think we can simply check for `if not db.result():` because a perfectly valid and successful query might return no data (for example, if there's a `where` clause with a condition that does not obtain anywhere in the data) and then the result is an empty list, which also evaluates as falsy. So, at a minimum I'd like to keep it convenient for the calling function to distinguish empty results from actual exceptions.

I guess for me all this boils down to, right now my thinking is: I want to keep your addition of logging the exception here, but then would also prefer to continue to return the serialized exception as before. I am honestly willing to entertain arguments to the contrary, if you have any interest in making them. There may be other ways to address my concerns, or the concerns themselves may be misplaced. Let me know what you think.